In [95]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [119]:
brady = pd.DataFrame()
brady['kids'] = ['Greg','Marcia','Peter','Jan','Bobby','Cindy','Oliver']
brady['ages'] = [14,12,11,10,8,6,8]
brady

kids  ages
0    Greg    14
1  Marcia    12
2   Peter    11
3     Jan    10
4   Bobby     8
5   Cindy     6
6  Oliver     8

In [117]:
mean = np.mean(brady['ages'])
median = np.median(brady['ages'])
(values, counts) = np.unique(brady['ages'], return_counts=True)
ind = np.argmax(counts)
mode = values[ind]
print('The mean age is {}. \nThe median age is {}. \nThe mode is {}.'.format(mean,median,mode))

The mean age is 9.0. 
The median age is 10.0. 
The mode is 1.


In [65]:
variance = brady['ages'].var()
sd = np.std(brady['ages'], ddof=1)
se = np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages']))
print('The variance is {}.\nThe standard deviation is {}.\nThe standard error is {}.'.format(variance,sd,se))

The variance is 7.47619047619.
The standard deviation is 2.73426232761.
The standard error is 1.03345401972.


For such a small data set, which doesn't include any extreme outliers, the mean is perhaps the most informative measure of central tendancy. The variance and the standard deviation are rather high, because the sample size is so small. In this case, we might report the standard deviation, to show that the data aren't normally distributed. 

In [120]:
brady.loc[brady.kids == 'Cindy', ['ages']] = brady['ages'] +1
mean = np.mean(brady['ages'])
median = np.median(brady['ages'])
(values, counts) = np.unique(brady['ages'], return_counts=True)
ind = np.argmax(counts)
mode = values[ind]
print('Now the mean age is {}. \nThe median age is {}. \nThe mode is {}.'.format(mean,median,mode))

variance = brady['ages'].var()
sd = np.std(brady['ages'], ddof=1)
se = np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages']))
print('The variance is {}.\nThe standard deviation is {}.\nThe standard error is {}.'.format(variance,sd,se))

Now the mean age is 10.0. 
The median age is 10.0. 
The mode is 8.
The variance is 6.33333333333.
The standard deviation is 2.51661147842.
The standard error is 0.951189731211.


When one year is added to Cindy's age, the mean and median increase. Because more values are now closer to the mean, the estimates of variance decrease a little bit. 

In [121]:
brady = brady[brady.kids != 'Oliver']
brady = brady.append(pd.DataFrame([[1, 'Jessica']], columns=list(['ages','kids'])))
print(brady)

mean = np.mean(brady['ages'])
median = np.median(brady['ages'])
(values, counts) = np.unique(brady['ages'], return_counts=True)
ind = np.argmax(counts)
mode = values[ind]
print('Now the mean age is {}. \nThe median age is {}. \nThere is no mode.'.format(mean,median))

variance = brady['ages'].var()
sd = np.std(brady['ages'], ddof=1)
se = np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages']))
print('The variance is {}.\nThe standard deviation is {}.\nThe standard error is {}.'.format(variance,sd,se))

   ages     kids
0    14     Greg
1    12   Marcia
2    11    Peter
3    10      Jan
4     8    Bobby
5     7    Cindy
0     1  Jessica
Now the mean age is 9.0. 
The median age is 10.0. 
There is no mode.
The variance is 18.0.
The standard deviation is 4.24264068712.
The standard error is 1.60356745147.


In [122]:
survey = pd.DataFrame()
survey['mags'] = ['TV Guide','EW','Pop Culture Today','SciPhi Phanatic']
survey['percents'] = [20,23,17,5]
survey

mags  percents
0           TV Guide        20
1                 EW        23
2  Pop Culture Today        17
3    SciPhi Phanatic         5

In [124]:
print(np.mean(survey['percents']))
print(np.median(survey['percents']))
print(np.std(survey['percents'] ,ddof=1) / np.sqrt(len(survey['percents'])))



16.25
18.5
3.94493345951


In [125]:
survey = survey[survey.mags != 'SciPhi Phanatic']
print(np.mean(survey['percents']))
print(np.median(survey['percents']))
print(np.std(survey['percents'] ,ddof=1) / np.sqrt(len(survey['percents'])))

20.0
20.0
1.73205080757


Given the mean and the standard error of the sample, we could estimate that 16.25% of adults were fans, give or take 3.94%. On the other hand, since the low percentage of respondents from the SciPhi Phanatic sample skews the mean, we could remove this outlier and recompute the mean. If we do so, the mean and the median are the same, suggesting that 20% is a more precise estimate of the population. 